### PR TITLE
Don't store custom names from TitleDatabase in GameFileCache

### PR DESCRIPTION
**Pull request by JosJuice**
This saves us from having to update the GameFileCache when the TitleDatabase changes (for instance when the user changes language).

### CODE REVIEW

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 1]
```diff
@@ -383,28 +393,34 @@ void GameListCtrl::RefreshList()
...
+  m_shown_names.clear();
   m_shown_files.clear();
   {
     std::unique_lock<std::mutex> lk(m_cache_mutex);
-    m_cache.ForEach([this](const std::shared_ptr<const UICommon::GameFile>& game_file) {
-      if (ShouldDisplayGameListItem(*game_file))
-        m_shown_files.push_back(game_file);
-    });
+    m_cache.ForEach(
+        [this, title_database](const std::shared_ptr<const UICommon::GameFile>& game_file) {
```
Comment (leoetlino) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: Did you intend to take the title database by const ref?

Comment (JosJuice) [reply Source/Core/DolphinWX/GameListCtrl.cpp]: Or, well, it's not possible to use a const ref in a lambda capture as far as I know... But I do want to take a ref.

Comment (leoetlino) [reply Source/Core/DolphinWX/GameListCtrl.cpp]: IIRC everything you capture into a lambda is implicitly const, unless you add mutable before the `{`.

Comment (leoetlino) [reply Source/Core/DolphinWX/GameListCtrl.cpp]: Hmm, just checked and it seems like I'm misremembering. However if you make title_database const it should have the same effect.

File: Source/Core/UICommon/GameFile.cpp [hunk 2]
```diff
@@ -323,9 +314,7 @@ std::string GameFile::GetUniqueIdentifier() const
   if (GetRevision() != 0)
     info.push_back("Revision " + std::to_string(GetRevision()));
 
-  std::string name(GetLongName(lang));
-  if (name.empty())
-    name = GetName();
+  const std::string& name = GetName();
```
Comment (leoetlino) [inline Source/Core/UICommon/GameFile.cpp]: This won't use the long name anymore AFAICT -- is this intended?

Comment (JosJuice) [reply Source/Core/UICommon/GameFile.cpp]: This has the same behavior as before. GetName will return the long name unless you pass in `false` (or unless there is no long name, in which case it returns the filename).

**Pull request merged**